src/garage/experiment/snapshotter.py

```diff
@@ -1,127 +1,125 @@
-                pass
-            else:
-                raise ValueError('Invalid snapshot mode {}'.format(
-                    self._snapshot_mode))
-
-            if file_name:
-                with open(file_name, 'wb') as file:
-                    pickle.dump(params, file)
-
-    def load(self, itr='last'):
-        r"""Load one snapshot of parameters from disk.
+            file_name_last = os.path.join(self._snapshot_dir, 'params.pkl')
+            with open(file_name_last, 'wb') as file:
+                pickle.dump(params, file)
+        elif self._snapshot_mode == 'none':
+            pass
+        else:
+            raise ValueError('Invalid snapshot mode {}'.format(
+                self._snapshot_mode))
+
+        if file_name:
+            with open(file_name, 'wb') as file:
+                pickle.dump(params, file)
+
+    def load(self, resume_from_dir, itr='last'):
+        """Load one snapshot of parameters from disk.
 
         Args:
-            itr(int or string): Iteration to load.
+            resume_from_dir (str): Directory of the pickle file
+                to resume experiment from.
+            itr (int or string): Iteration to load.
                 Can be an integer, 'last' or 'first'.
 
         Returns:
             dict: Loaded snapshot
 
         """
         if isinstance(itr, int) or itr.isdigit():
-            snapshot_file = osp.join(self._snapshot_dir,
-                                     'itr_{}.pkl'.format(itr))
+            resume_from_file = os.path.join(resume_from_dir,
+                                            'itr_{}.pkl'.format(itr))
         else:
             if itr not in ('last', 'first'):
                 raise ValueError(
                     "itr should be an integer or 'last' or 'first'")
 
-            snapshot_file = osp.join(self._snapshot_dir, 'params.pkl')
-            if not osp.isfile(snapshot_file):
+            resume_from_file = os.path.join(resume_from_dir, 'params.pkl')
+            if not os.path.isfile(resume_from_file):
                 files = [
-                    f for f in listdir(self._snapshot_dir)
+                    f for f in os.listdir(resume_from_dir)
                     if f.endswith('.pkl')
                 ]
                 if not files:
                     raise FileNotFoundError(errno.ENOENT,
                                             os.strerror(errno.ENOENT),
                                             '*.pkl file')
                 files.sort(key=lambda f: int(re.findall(r'\d+', f)[0]))
-                snapshot_file = files[0] if itr == 'first' else files[-1]
-                snapshot_file = osp.join(self._snapshot_dir, snapshot_file)
+                resume_from_file = files[0] if itr == 'first' else files[-1]
+                resume_from_file = os.path.join(resume_from_dir,
+                                                resume_from_file)
 
-        assert osp.isfile(snapshot_file)
+        assert os.path.isfile(resume_from_file)
```

Comment:
if you are detecting errors from user input, raise an exception, don't use assert.